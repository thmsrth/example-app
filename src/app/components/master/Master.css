.Master {
  box-shadow: var(--section-box-shadow);
  background-color: var(--section-background-color);
  border-top: var(--section-border-top);
  border-radius: var(--section-border-radius);
  margin: var(--section-margin);
}

.Master__container {
  padding: var(--section-padding);
}

.Master__actions {
  padding: calc(var(--section-padding) * 2) var(--section-padding) 0;
}

.Master__header {
  width: 100%;
  padding: var(--section-header-padding);
  border-bottom: var(--section-header-border-bottom);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.Master__title {
  font-size: var(--section-title-font-size);
  margin: var(--section-title-margin);
}
