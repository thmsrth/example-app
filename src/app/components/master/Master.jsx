import React from 'react';

import PropTypes from 'prop-types';
import dataItemType from '../types/dataItem';

import I18n from '../../utils/I18n';

import DataStateNotifier from '../../../global/components/DataStateNotifier';

import MasterList from './MasterList';
import MasterFilter from './MasterFilter';
import MasterActions from './MasterActions';
import MasterPagination from './MasterPagination';

import './Master.css';

const Master = ({
  handleFilterItems, filteredData, ...otherProps
}) => (
  <DataStateNotifier {...otherProps}>
    <div className="Master">
      <div className="Master__header">
        <h3 className="Master__title">
          {I18n.t('en.data.title')}
        </h3>
      </div>
      <div className="Master__container">
        <MasterFilter
          {...otherProps}
          handleFilterItems={handleFilterItems}
        />
        <MasterActions
          {...otherProps}
        />
        <MasterList
          {...otherProps}
          data={filteredData}
        />
        <MasterPagination
          {...otherProps}
          data={filteredData}
        />
      </div>
    </div>
  </DataStateNotifier>
);

export default Master;

Master.propTypes = {
  filteredData: PropTypes.arrayOf(dataItemType),
  handleFilterItems: PropTypes.func,
};

Master.defaultProps = {
  filteredData: [],
  handleFilterItems: () => {},
};
