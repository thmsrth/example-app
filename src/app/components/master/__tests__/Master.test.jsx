import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import staticData from '../../../static/data.json';
import Master from '../Master';

global.permissions = {
  EXAMPLE_LIST: true,
};

const { _embedded } = staticData;

const props = {
  data: _embedded.foods,
  dataLoading: '',
  dataError: '',
};

describe('Master', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<Master {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
