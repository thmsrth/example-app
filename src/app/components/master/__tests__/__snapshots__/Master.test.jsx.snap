// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Master should render the component correctly 1`] = `
<DataStateNotifier
  data={
    Array [
      Object {
        "calories": 266,
        "createdAt": "2018-10-28T14:48:54.01293",
        "createdBy": "thomas",
        "fat": 34,
        "id": 1,
        "name": "Pizza",
        "protein": 0.6,
        "status": "UNHEALTHY",
        "updatedAt": null,
        "updatedBy": null,
      },
      Object {
        "calories": 508,
        "createdAt": "2018-11-28T04:48:54.01293",
        "createdBy": "tommy",
        "fat": 1,
        "id": 2,
        "name": "Sushi",
        "protein": 26.7,
        "status": "HEALTHY",
        "updatedAt": null,
        "updatedBy": null,
      },
      Object {
        "calories": 271,
        "createdAt": "2018-12-28T14:48:54.01293",
        "createdBy": "ben",
        "fat": 1.3,
        "id": 3,
        "name": "Steak",
        "protein": 6.6,
        "status": "UNHEALTHY",
        "updatedAt": "2018-12-29T12:48:54.01293",
        "updatedBy": "thomas",
      },
    ]
  }
  dataAdding=""
  dataDeleting=""
  dataError=""
  dataLoading=""
  dataUpdating=""
  resetDataError={[Function]}
>
  <div
    className="Master"
  >
    <div
      className="Master__header"
    >
      <h3
        className="Master__title"
      >
        Food
      </h3>
    </div>
    <div
      className="Master__container"
    >
      <MasterFilter
        data={
          Array [
            Object {
              "calories": 266,
              "createdAt": "2018-10-28T14:48:54.01293",
              "createdBy": "thomas",
              "fat": 34,
              "id": 1,
              "name": "Pizza",
              "protein": 0.6,
              "status": "UNHEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            },
            Object {
              "calories": 508,
              "createdAt": "2018-11-28T04:48:54.01293",
              "createdBy": "tommy",
              "fat": 1,
              "id": 2,
              "name": "Sushi",
              "protein": 26.7,
              "status": "HEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            },
            Object {
              "calories": 271,
              "createdAt": "2018-12-28T14:48:54.01293",
              "createdBy": "ben",
              "fat": 1.3,
              "id": 3,
              "name": "Steak",
              "protein": 6.6,
              "status": "UNHEALTHY",
              "updatedAt": "2018-12-29T12:48:54.01293",
              "updatedBy": "thomas",
            },
          ]
        }
        dataError=""
        dataFilters={Object {}}
        dataLoading=""
        fetchFilteredItems={[Function]}
        filters={
          Array [
            Object {
              "name": "id",
              "title": "ID:",
              "type": "number",
            },
            Object {
              "name": "name",
              "title": "Name:",
              "type": "text",
            },
            Object {
              "dateMax": "2099-12-31",
              "dateMin": "2012-01-01",
              "name": "createdAt",
              "pattern": "[0-9]{4}-[0-9]{2}-[0-9]{2}",
              "placeholder": "yyyy-mm-dd",
              "range": true,
              "title": "Created At:",
              "type": "datetime",
            },
            Object {
              "name": "createdBy",
              "title": "Created by:",
              "type": "text",
            },
            Object {
              "dateMax": "2099-12-31",
              "dateMin": "2012-01-01",
              "name": "updatedAt",
              "pattern": "[0-9]{4}-[0-9]{2}-[0-9]{2}",
              "placeholder": "yyyy-mm-dd",
              "range": true,
              "title": "Updated At:",
              "type": "datetime",
            },
            Object {
              "name": "updatedBy",
              "title": "Updated by:",
              "type": "text",
            },
          ]
        }
        handleFilterItems={[Function]}
      />
      <MasterActions
        data={
          Array [
            Object {
              "calories": 266,
              "createdAt": "2018-10-28T14:48:54.01293",
              "createdBy": "thomas",
              "fat": 34,
              "id": 1,
              "name": "Pizza",
              "protein": 0.6,
              "status": "UNHEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            },
            Object {
              "calories": 508,
              "createdAt": "2018-11-28T04:48:54.01293",
              "createdBy": "tommy",
              "fat": 1,
              "id": 2,
              "name": "Sushi",
              "protein": 26.7,
              "status": "HEALTHY",
              "updatedAt": null,
              "updatedBy": null,
            },
            Object {
              "calories": 271,
              "createdAt": "2018-12-28T14:48:54.01293",
              "createdBy": "ben",
              "fat": 1.3,
              "id": 3,
              "name": "Steak",
              "protein": 6.6,
              "status": "UNHEALTHY",
              "updatedAt": "2018-12-29T12:48:54.01293",
              "updatedBy": "thomas",
            },
          ]
        }
        dataError=""
        dataLoading=""
      />
      <MasterList
        data={Array []}
        dataError=""
        dataLoading=""
        dataPagination={
          Object {
            "number": 0,
            "numberOfElements": 0,
            "size": 25,
            "totalElements": 0,
            "totalPages": 1,
          }
        }
        dataSort={
          Object {
            "direction": "DESC",
            "property": "createdAt",
          }
        }
        fetchSortedItems={[Function]}
      />
      <MasterPagination
        data={Array []}
        dataError=""
        dataLoading=""
        dataPageSize={25}
        dataPagination={
          Object {
            "number": 0,
            "numberOfElements": 0,
            "size": 25,
            "totalElements": 0,
            "totalPages": 1,
          }
        }
        dataSortParam="id,ASC"
        onPageNumberChange={[Function]}
        onSizeChange={[Function]}
      />
    </div>
  </div>
</DataStateNotifier>
`;
